nucliadb/nucliadb/ingest/orm/resource.py

```diff
@@ -358,10 +358,16 @@ async def generate_index_message(self) -> ResourceBrain:
                         ),
                         None,
                     )
+                if reindex:
+                    replace_field = [
+                        f"{p.start}-{p.end}" for p in field_metadata.metadata.paragraphs
+                    ]
+                else:
+                    replace_field = []
                 brain.apply_field_metadata(
                     field_key,
                     field_metadata,
-                    replace_field=[],
+                    replace_field=replace_field,
```

Comment:
rename to `paragraphs_to_replace` ?
Do we really need this though? AFAIK we only need to know the key prefix up to the field part

Reply:
We actually need it, the node expect a paragraph id. Changing this to work by field makes sense but it's a bit more involved, so I wanted to avoid it in this PR. I'll change the name!